src/framework/mpas_dmpar.F

```diff
@@ -6167,6 +6180,197 @@ subroutine mpas_dmpar_exch_halo_field5d_real(field, haloLayersIn)!{{{
 
    end subroutine mpas_dmpar_exch_halo_field5d_real!}}}
 
+   !-----------------------------------------------------------------------
+   !  routine mpas_dmpar_exch_halo_adj_field2d_real
+   !
+   !> \brief MPAS dmpar halo exchange adjoint 2D real field
+   !> \author BJ Jung
+   !> \date   09/2020
+   !> \details
+   !>  This routine handles the adjoint of halo exchange communication of an input field across all processors.
+   !>  It accumulates the values of owned point with the values of halos. It is based on mpas_dmpar_exch_halo_field2d_real.
+   !
+   !> Note the number of halo layers impacts the number of cells which will be updated by this routine:
+   !< The first halo layer will update the owned 'edge' cells, where 'edge' cells are adjacent to qhost cells.
+   !< The second halo layer will update owned cells which are adjacent to the 'edge' cells.
+   !< The third halo layer will update owned cells which are adjacent to the cells updated by the seconds halo layer, etc.
```

Comment:
@mgduda @byoung-joo Do these comments make sense? Is this a good place to document this?

Should I add this to the commit message for the commit to add the tests? Should I add it to the PR description?